pallets/staking_rewards/src/tests.rs

```diff
@@ -273,27 +317,159 @@ fn test_yearly_rewards() {
         t(10_000, 10_000 / 10);
 
         // Decay is set to 20%
-        <Test as Config>::RewardDecayPct::set(Percent::from_percent(20));
+        <Test as Config>::LowRateRewardDecayPct::set(Percent::from_percent(20));
         t(10_000, 10_000 / 5);
 
         // Decay is set to 25%
-        <Test as Config>::RewardDecayPct::set(Percent::from_percent(25));
+        <Test as Config>::LowRateRewardDecayPct::set(Percent::from_percent(25));
         t(10_000, 10_000 / 4);
 
         // Decay is set to 50%
-        <Test as Config>::RewardDecayPct::set(Percent::from_percent(50));
+        <Test as Config>::LowRateRewardDecayPct::set(Percent::from_percent(50));
         t(10_000, 10_000 / 2);
     })
 }
 
+#[test]
+fn test_yearly_rewards_high_rate() {
+    // Test yearly rewards at different staking rates
+    new_test_ext().execute_with(|| {
+        fn t(emission_supply: u64, max_yearly_decay: u64) {
```

Comment:
This function seems exactly like `t` from the previous test except for the `on_runtime_upgrade`. Can you de-duplicate this?